crates/bitwarden-uuid-macro/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, Ident, Visibility,
+};
+
+#[proc_macro]
+pub fn uuid(input: TokenStream) -> TokenStream {
+    // Parse input as: vis ident
+    let input = parse_macro_input!(input as IdTypeInput);
+    let ident = input.ident;
+    let vis = input.vis;
+    let name_str = ident.to_string();
+
+    let tsify_type = format!("Tagged<Uuid, \"{}\">", name_str);
+    let doc_string = format!(" NewType wrapper for `{}`", name_str);
+
+    let expanded = quote! {
+        #[doc = #doc_string]
+        #[cfg_attr(feature = "wasm", derive(::tsify_next::Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+        #[derive(::serde::Serialize, ::serde::Deserialize)]
```

Comment:
Do we know if just having `#[derive(::serde::Serialize, ::serde::Deserialize)]` and `#[repr(transparent)]` makes serde serialize it as a string, instead of a struct with one field? I know serde has it's own `#[serde(transparent)]`, but not sure if it's needed here. May be a good idea to add a serialization/deserialization test just in case.

It's also a good idea to derive Clone and Copy for this, potentially also Eq,Hash if we ever want to use them as keys in a hashmap

Reply:
Seems towork.